webapp/src/plugin.jsx

```diff
@@ -42,6 +43,38 @@ function getTranslations(locale) {
     return {};
 }
 
+// eslint-disable-next-line no-unused-vars
+function addToAdminConsole(registry) {
+    registry.registerAdminConsolePlugin((adminDefinition) => {
+        // the settings will appear on the authentication section
+        const demoConfig = {
+            url: 'plugins/demo',
+            icon: 'fa-rocket',
+            title: 'com.mattermost.demo-plugin.title',
+            title_default: 'Demo plugin admin',
+            isHidden: () => false,
+            schema: {
+                id: 'DemoPlugin',
+                component: AdminSettings,
+            },
+        };
+
+        // this is not required - demoConfig could be embedded elsewhere
+        adminDefinition.pluginsExtra = {
```

Comment:
I'm not sure if this is the approach that we want to promote. maybe we prefer something like:

```js
adminDefinition.plugins['demo'] = demoConfig
```

Reply:
If I'm recalling correctly, currently the plugins section is special, and so that wouldn't work. I agree though, and believe we should perhaps confine the plugins admin consoles to this section after it accepts custom consoles. Additionally, those with custom sections should probably have the default section (from the schema/config) inhibited.

https://github.com/mattermost/mattermost-webapp/blob/master/components/admin_console/admin_sidebar/admin_sidebar.jsx#L204

Reply:
The idea of have this "full access" is because we want super powerful plugins, if you don't want to use a plugin, you can just don't use it, but if you use a plugin, you must know that it can do a lot of things. In this case, edit entirely the admin console. For example, I would like to create a plugin that hides sections, so... you can simplify the admin console removing enything that is "distracting for you".

About the plugin sections special case, the plugins section will always render the plugins config items, but you can still add things to it, because it will show the items in the admin definitions too.